core/src/main/java/org/elasticsearch/action/admin/cluster/node/tasks/list/TransportListTasksAction.java

```diff
@@ -63,6 +71,33 @@ protected TaskInfo taskOperation(ListTasksRequest request, Task task) {
     }
 
     @Override
+    protected void processTasks(ListTasksRequest request, Consumer<Task> operation) {
+        if (false == request.shouldWaitForCompletion()) {
+            super.processTasks(request, operation);
+            return;
+        }
+        // If we should wait for completion then we have to intercept every found task and wait for it to leave the manager.
+        TimeValue timeout = request.timeout();
+        if (timeout == null) {
+            timeout = DEFAULT_WAIT_FOR_COMPLETION_TIMEOUT;
+        }
+        long timeoutTime = System.nanoTime() + request.timeout().nanos();
+        super.processTasks(request, operation.andThen((Task t) -> {
+            while (System.nanoTime() < timeoutTime) {
```

Comment:
Documentation for nanoTime() is suggesting to use `System.nanoTime() - timeoutTime < 0` instead to avoid numerical overflow.